### PR TITLE
RevEng: Moving and simplifying look up of assembly from name

### PR DESCRIPTION
**Pull request by lajones**
@bricelam - just tidying up so that we pass the assembly name and look up the actual assembly in DatabaseTool instead. As agreed earlier today.

### CODE REVIEW

File: src/EntityFramework.Commands/Properties/Strings.resx [hunk 1]
```diff
@@ -60,122 +59,119 @@
...
     <value>The current CSharpMigrationOperationGenerator cannot scaffold operations of type '{operationType}'. Configure your services to use one that can.</value>
   </data>
-    <data name="UnknownLiteral" xml:space="preserve">
+  <data name="UnknownLiteral" xml:space="preserve">
     <value>The current CSharpHelper cannot scaffold literals of type '{literalType}'. Configure your services to use one that can.</value>
   </data>
+  <data name="CannotFindAssembly" xml:space="preserve">
+    <value>Unable to find asssembly with name {assemblyName}.</value>
+  </data>
 </root>
```
Comment (lajones) [inline src/EntityFramework.Commands/Properties/Strings.resx]: Not sure why all the whitespace changes. The 3 lines above are the only real change.

File: src/EntityFramework.Commands/DatabaseTool.cs [hunk 2]
```diff
@@ -27,26 +30,33 @@ public class DatabaseTool
...
-            Check.NotNull(providerAssembly, nameof(providerAssembly));
+            Check.NotNull(providerAssemblyName, nameof(providerAssemblyName));
             Check.NotEmpty(connectionString, nameof(connectionString));
             Check.NotEmpty(rootNamespace, nameof(rootNamespace));
             Check.NotEmpty(projectDir, nameof(projectDir));
 
+            var assembly = GetReverseEngineerProviderAssembly(providerAssemblyName);
+            if (assembly == null)
+            {
+                _logger.Value.LogWarning(Strings.CannotFindAssembly(providerAssemblyName));
```
Comment (bricelam) [inline src/EntityFramework.Commands/DatabaseTool.cs]: You can always throw exceptions to "log" errors.

File: src/EntityFramework.Commands/Properties/Strings.resx (hunk 1 above)
Comment (bricelam) [inline src/EntityFramework.Commands/Properties/Strings.resx]: Someone probably ran R# Code Clean-up which formats it differently than VS does by default.

**Comment by bricelam** [comment]
:shipit:

**Comment by bricelam** [comment]
Can we remove `Microsoft.SqlServer.Design` from this project's dependencies now?

File: src/EntityFramework.Commands/DatabaseTool.cs (hunk 2 above)
Comment (lajones) [inline src/EntityFramework.Commands/DatabaseTool.cs]: Did it this was to be consistent with some logging that MigrationTool does. I don't really mind which way though. Would you prefer exceptions? Any particular kind of exception?

**Comment by lajones** [comment]
Re  'Microsoft.SqlServer.Design'  - it's still hard-coded to use that provider and only that provider. This was per agreement with @rowanmiller a while back. In future will want to add code to the psm to allow using the DTE project to pass in the project's references and using those to look up the provider assembly but that's on the back-burner for right now.

File: src/EntityFramework.Commands/DatabaseTool.cs (hunk 2 above)
Comment (lajones) [inline src/EntityFramework.Commands/DatabaseTool.cs]: Looks like ContextTool uses InvalidOperationException. So I'll do the same.

**Comment by lajones** [comment]
Checked in with commit c2e146a.

**Pull request closed without merging**